modules/github_integration/lib/api/v3/github_integration/github_pull_request_representer.rb

```diff
@@ -0,0 +1,166 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+require 'roar/decorator'
+require 'roar/json/hal'
+
+module API
+  module V3
+    module GithubIntegration
+      class GithubPullRequestRepresenter < ::API::Decorators::Single
+        include API::Caching::CachedRepresenter
+        include API::Decorators::DateProperty
+        include API::Decorators::LinkedResource
+
+        cached_representer key_parts: %i[github_user merged_by latest_check_runs],
+                           disabled: false
+
+        link :staticPath do
+          next
+        end
+
+        property :id
+
+        property :number,
+                 setter: ->(*) { nil }
+
+        property :github_html_url,
+                 setter: ->(*) { nil }
+
+        property :state,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :repository,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        date_time_property :github_updated_at,
+                           render_nil: true,
+                           setter: ->(*) { nil }
+
+        property :title,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :body,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :draft,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged_at,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :review_comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :additions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :deletions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :changed_files_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :labels,
+                 setter: ->(*) { nil }
+
+        property :github_user,
+                 getter: ->(*) {
+                   {
+                     login: github_user.github_login,
+                     htmlUrl: github_user.github_html_url,
+                     avatarUrl: github_user.github_avatar_url
+                   }
+                 },
+                 setter: ->(*) { nil }
```

Comment:
The way the OpenProject api works, the GithubUser would be a resources in its own right and the GithubPullRequest resource would `link` to it.

Reply:
not exactly sure how to best approach that (although I agree that would be better), see https://github.com/opf/openproject/pull/9070#discussion_r611972158